composer/blog/gcp-tech-blog/data-orchestration-with-composer/dags/data_orchestration_blog_sample_dag.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from airflow.utils import dates
+from airflow import models
+from airflow.hooks.base_hook import BaseHook
+from airflow.utils.state import State
+from airflow.providers.google.cloud.sensors.gcs import GCSObjectExistenceSensor
+from airflow.providers.google.cloud.operators.dataflow import DataflowTemplatedJobStartOperator
+from airflow.providers.google.cloud.operators.gcs import GCSDeleteBucketOperator
+from airflow.providers.google.cloud.operators.bigquery import BigQueryCheckOperator
+from airflow.contrib.operators.slack_webhook_operator import SlackWebhookOperator
+
+BUCKET_NAME = models.Variable.get("bucket_name")
+FILE_NAME = models.Variable.get("file_name")
+BQ_SQL_STRING = models.Variable.get("bq_sql_string")
+
+def on_failure_callback(context):
+    ti = context.get('task_instance')
+    slack_msg = """
+            :red_circle: Task Failed. 
+            *Task*: {task}  
+            *Dag*: {dag} 
+            *Execution Time*: {exec_date}  
+            *Log Url*: {log_url} 
+            """.format(
+            task=ti.task_id,
+            dag=ti.dag_id,
+            log_url=ti.log_url,
+            exec_date=context.get('execution_date'),
+        )
+    slack_webhook_token = BaseHook.get_connection('slack_connection').password
```

Comment:
I didn't see setting the Slack connection as part of the blog post - I'm guessing they need to have that configured in Airflow > Connections? Might be worth adding as part of the setup. You may also want to consider adding a link to the relevant Airflow docs about configuring this connection in the comments here, but that's a "nice to have" vs a "totally necessary"

Reply:
added the link to a tutorial on how to set this up and I've added a one liner to the blog post covering this